### ISSUE TITLE
Download stuck when a request fails

### ISSUE THREAD

**Issue by DFFZMXJ** [MinhasKamal/DownGit#28]
In #27 , I said I encountered the same problem as that guy. Actually, as I observe, it seems a little different. When my download progress is stuck, I can see this in F12:

![image](https://user-images.githubusercontent.com/35987451/72134019-415de580-3359-11ea-9bd9-6ed6b0fcb0be.png)

Its response:

```json
{
  "message": "API rate limit exceeded for 45.32.27.8. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
  "documentation_url": "https://developer.github.com/v3/#rate-limiting"
}
```

Your program requests too frequently but won't retry when failing. Therefore the download will fail even if only one error occurs. I suggest retrying for 3 times at least or set a button.

**Comment by luki1412** [comment]
Im getting: 
**Failed to load resource: net::ERR_INSUFFICIENT_RESOURCES**
and
**Failed to load resource: the server responded with a status of 403 (Forbidden)**

UI/web page should also display an error of some sort.

**Comment by MinhasKamal** [comment]
Yes, some sort of error message would help. I'll try to get into it as soon as I can.

**Comment by Metroite** [comment]
I think its related to this: If the target is too big, it gets stuck and loads indefinitely. On GitZip it works. I assume its because of the **Github API Access Token**.
